custom-wasm-transformers/samples/wasm-java-helm-chart/add-custom-files-dirs/main.go

```diff
@@ -0,0 +1,272 @@
+	pathTemplate := "{{ SourceRel .ServiceFsPath }}"
+	for _, v := range newArtifacts {
+		_serviceName, ok := v.Configs["Service"].(map[string]interface{})
+		if !ok {
+			fmt.Println("failed to v.Configs[Service].(map[string]interface{})")
+			continue
+		}
+		// serviceName, ok := _serviceName["serviceName"].(string)
+		serviceName, ok := _serviceName["ServiceName"].(string)
+		if !ok {
+			fmt.Printf("failed to _serviceName[ServiceName].(string). actual _serviceName: %T %+v\n", _serviceName, _serviceName)
+			continue
+		}
+		if len(v.Paths) == 0 {
+			fmt.Printf("failed to v.Paths. actual v.Paths: %T %+v\n", v.Paths, v.Paths)
+			continue
+		}
+		projPaths := v.Paths["ProjectPath"]
+		if len(projPaths) == 0 {
+			fmt.Printf("failed to v.Paths[ProjectPath]. actual v.Paths: %T %+v\n", projPaths, projPaths)
+			continue
+		}
+		fmt.Printf("len(projPaths) %d %+v\n", len(projPaths), projPaths)
+		dir := projPaths[0]
+		// Create a path template for the service
+		pathTemplateName := strings.ReplaceAll(serviceName, "-", "") + "path"
+		tplPathData := map[string]interface{}{
+			"ServiceFsPath":    dir,
+			"PathTemplateName": pathTemplateName,
+		}
+		pathMappings = append(pathMappings,
+			PathMapping{
+				Type:           "PathTemplate",
+				SrcPath:        pathTemplate,
+				TemplateConfig: tplPathData,
+			})
+		// Since the helm chart uses the same templating character {{ }} as Golang templates,
+		// we use `SpecialTemplate` type here where the templating character is <~ ~>.
+		// The `Template` type can be used for all normal cases
+		pathMappings = append(pathMappings, PathMapping{
+			Type:     "SpecialTemplate",
+			DestPath: "{{ ." + pathTemplateName + " }}",
+			TemplateConfig: map[string]interface{}{
+				"ServiceFsPath": dir,
+				"ServiceName":   serviceName,
+			}})
+		pathMappings = append(pathMappings, PathMapping{
+			Type:     "Source",
+			SrcPath:  "{{ ." + pathTemplateName + " }}",
+			DestPath: "{{ ." + pathTemplateName + " }}",
+		})
+	}
+	return pathMappings, artifacts, nil
+}
+
+//go:export RunDirectoryDetect
+func RunDirectoryDetect(
+	inputJsonPtr uint32,
+	inputJsonLen uint32,
+	outputJsonPtr uint32,
+	outputJsonLen uint32,
+) int32 {
+	fmt.Println("mycustomtransformer: RunDirectoryDetect start")
+	defer fmt.Println("mycustomtransformer: RunDirectoryDetect end")
+	transformInputJson := ptrToString(inputJsonPtr, inputJsonLen)
+	input := DirDetectInput{}
+	if err := json.Unmarshal([]byte(transformInputJson), &input); err != nil {
+		fmt.Printf("mycustomtransformer: failed to unmarshal input. Error: %q\n", err)
+		return -1
+	}
+	ps, err := DirectoryDetect(input.SourceDir)
+	if err != nil {
+		fmt.Printf("mycustomtransformer: failed to transform. Error: %q\n", err)
+		return -1
+	}
+	output := DirDetectOutput{Services: ps}
+	outputJson, err := json.Marshal(output)
+	if err != nil {
+		fmt.Println("mycustomtransformer: failed to marshal")
+		// panic("mycustomtransformer: failed to marshal")
+		return -1
```

Comment:
return values (-1) could be given a name.

Reply:
done

Reply:
I still see the old code. Is this change pushed?

Reply:
```I still see the old code. Is this change pushed?```

Yes I have added a name for the return value in the function signature.

Reply:
Yeah I pushed the changes but there's some weirdness with Github when you click the link above.

Use this link https://github.com/konveyor/move2kube-transformers/pull/56/files#diff-8685aa66a26bcc06417023f9c92402da887eda580b216cc3b1012b5d0ff3145fR229-R234